src/Vulkan/Silk.NET.Vulkan/Vk.cs

```diff
@@ -167,31 +168,34 @@ public unsafe bool IsInstanceExtensionPresent(string extension, string layer = n
             {
                 var layerPtr = layer == null ? 0 : SilkMarshal.StringToPtr(layer);
 
-                // Get count of properties
-                var instanceExtPropertiesCount = 0u;
-                EnumerateInstanceExtensionProperties((byte*) layerPtr, &instanceExtPropertiesCount, null);
+                try {
+                    // Get count of properties
+                    var instanceExtPropertiesCount = 0u;
+                    EnumerateInstanceExtensionProperties((byte*) layerPtr, &instanceExtPropertiesCount, null);
 
-                // Initialise return structure
-                using var mem = GlobalMemory.Allocate((int) instanceExtPropertiesCount * sizeof(ExtensionProperties));
-                var props = (ExtensionProperties*) Unsafe.AsPointer(ref mem.GetPinnableReference());
+                    // Initialise return structure
+                    var props = stackalloc ExtensionProperties[(int)instanceExtPropertiesCount];
```

Comment:
This may be slightly nitpicky, but the extension property count is easily controlled externally and even for spec-compliant implementations there's not a low enough limit.
`ExtensionProperties` is additionally of significant size (~260 bytes).
I have seen implementations expose > 100 extensions, which likely isn't enough to outright crash the process, but depending on the setup still isn't ideal.
A misbehaving, or (highly unlikely) attacking implicit layer could easily bump this to numbers far to great for a stack allocation.
The benefit of this also seems very small so please elaborate or revert this change.

Reply:
i had misread what perksey had said [here](https://discord.com/channels/521092042781229087/587346162802229298/1127352316857688095)